src/test/1_OracleHub.t.sol

```diff
@@ -1081,4 +1124,243 @@ contract OracleHubTest is Test {
+    //Oracle failsafe tests
+    //
+    function testRevert_decommissionOracle_notInHub(address sender, address oracle) public {
+        vm.assume(oracle != address(oracleEthToUsd));
+        vm.assume(oracle != address(oracleLinkToUsd));
+        vm.assume(oracle != address(oracleSnxToEth));
+
+        vm.startPrank(sender);
+        vm.expectRevert("OH_DO: Oracle not in Hub");
+        oracleHub.decommissionOracle(oracle);
+        vm.stopPrank();
+    }
+
+    function testSuccess_decommissionOracle_NonExistingContract(address sender) public {
+        RevertingOracle revertingOracle = new RevertingOracle();
+
+        vm.prank(creatorAddress);
+        oracleHub.addOracle(
+            OracleHub.OracleInformation({
+                oracleUnit: 10 ** 18,
+                baseAssetBaseCurrency: uint8(Constants.UsdBaseCurrency),
+                quoteAsset: "REVERT",
+                baseAsset: "USD",
+                oracle: address(revertingOracle),
+                quoteAssetAddress: address(0),
+                baseAssetIsBaseCurrency: true,
+                isActive: true
+            })
+        );
+
+        vm.prank(sender);
+        oracleHub.decommissionOracle(address(revertingOracle));
+
+        (bool isActive,,,,,,,) = oracleHub.oracleToOracleInformation(address(revertingOracle));
+        assertEq(isActive, false);
+
+        address[] memory oracles = new address[](1);
+        oracles[0] = address(revertingOracle);
+
+        (uint256 rateInUsd, uint256 rateInBaseCurrency) = oracleHub.getRate(oracles, Constants.UsdBaseCurrency);
+
+        assertEq(rateInUsd, 0);
+        assertEq(rateInBaseCurrency, 0);
+    }
+
+    function testSuccess_decommissionOracle_answerTooLow(address sender) public {
+        vm.startPrank(creatorAddress);
+        oracleHub.addOracle(
+            OracleHub.OracleInformation({
+                oracleUnit: 10 ** 18,
+                baseAssetBaseCurrency: uint8(Constants.EthBaseCurrency),
+                quoteAsset: "SNX",
+                baseAsset: "ETH",
+                oracle: address(oracleSnxToEth),
+                quoteAssetAddress: address(snx),
+                baseAssetIsBaseCurrency: true,
+                isActive: true
+            })
+        );
+        oracleHub.addOracle(
+            OracleHub.OracleInformation({
+                oracleUnit: 10 ** 18,
+                baseAssetBaseCurrency: uint8(Constants.UsdBaseCurrency),
+                quoteAsset: "ETH",
+                baseAsset: "USD",
+                oracle: address(oracleEthToUsd),
+                quoteAssetAddress: address(eth),
+                baseAssetIsBaseCurrency: true,
+                isActive: true
+            })
+        );
+        vm.stopPrank();
+
+        vm.warp(2 weeks); //to not run into an underflow
+
+        vm.startPrank(oracleOwner);
+        //minAnswer is set to 100 in the oracle mocks
+        oracleSnxToEth.transmit(int256(1));
+        oracleEthToUsd.transmit(int256(500000000000));
```

Comment:
What's the point of this oracle? Should we also try to decommission it?

Reply:
just checking whether it works if only one of the oracles in the train is decomissioned